### ISSUE TITLE
Can't proxy to a server presenting a certificate with an IP address in its common name

### ISSUE THREAD

**Issue by lc-guy** [svenstaro/proxyboi#240]
If I try to proxy to a local ip with an https server running, e.g.  `https://1.2.3.4`, proxyboi crashes with the following message:

```
thread 'actix-rt:worker:0' panicked at 'rustls currently only handles hostname-based connections. See https://github.com/briansmith/webpki/issues/54: InvalidDNSNameError', /cargo/registry/src/github.com-1ecc6299db9ec823/actix-connect-2.0.0/src/ssl/rustls.rs:107:14
```

For what it's worth, rustls forked the webpki repo and is including the fix in the upcoming 0.21.0 release (https://github.com/rustls/rustls/issues/184) -- so you might want to rebuild with that soon

**Comment by svenstaro** [comment]
Yeah, we can't do IPs with TLS right now. Good to hear that this might be changing soon!

**Comment by lc-guy** [comment]
Another issue I have.

I added an entry to `/etc/hosts` to bypass this and used --insecure, but I get the `[ERROR] Internal Server Error: SendRequestError(Connect(Io(Custom { kind: InvalidData, error: WebPKIError(BadDER) })))` error no matter what. Isn't the flag supposed to ignore this?
